packages/schemas/src/workflow-context.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Workflow context injection utilities
+ * Provides functionality to inject pre-context steps into workflows dynamically
+ */
+import type { WorkflowAgentStep } from './workflow/types.js';
+
+/**
+ * Data structure for previous iteration information
+ */
+export interface PreviousIteration {
+  number: number;
+  title?: string;
+  description?: string;
+  changes?: string;
+}
+
+/**
+ * Context data for building pre-context steps
+ */
+export interface PreContextData {
+  /** The task ID */
+  taskId: string;
+  /** Initial task information */
+  initialTask: {
+    title: string;
+    description: string;
+  };
+  /** Previous iterations (if any) */
+  previousIterations?: PreviousIteration[];
+  /** Current (last) iteration being executed */
+  currentIteration?: PreviousIteration;
+}
```

Comment:
Btw, I noticed this file is present in the `rover-schemas` packages. In that case, I would follow the [same approach we are using for other resources](https://github.com/endorhq/rover/blob/main/docs/schemas.md): Zod schemas, intherit types, and a manager class.

You can ask Rover to refactor all this based on the `schemas.md` documentation.